internal/api/api.go

```diff
@@ -120,7 +121,95 @@ func handleHealthcheck(c *gin.Context) {
 	c.JSON(200, gin.H{"failed": false})
 }
 
+// Path parameters for GET requests
+type TxHashPathParams struct {
+	TxHash string `uri:"tx_hash" binding:"required"` // Transaction hash
+}
+
+// handleHasTx godoc
+//
+//	@Summary		HasTx
+//	@Description	Determine if a given transaction ID exists in the node mempool.
+//	@Produce		json
+//	@Param			tx_hash	path		string	true	"Transaction Hash"
+//	@Success		200		{object}	string	"Ok"
+//	@Failure		400		{object}	string	"Bad Request"
+//	@Failure		404		{object}	string	"Not Found"
+//	@Failure		415		{object}	string	"Unsupported Media Type"
+//	@Failure		500		{object}	string	"Server Error"
+//	@Router			/api/hastx/{tx_hash} [get]
+func handleHasTx(c *gin.Context) {
+	// First, initialize our configuration and loggers
+	cfg := config.GetConfig()
+	logger := logging.GetLogger()
+
+	var uriParams TxHashPathParams
+	if err := c.ShouldBindUri(&uriParams); err != nil {
+		logger.Errorf("failed to bind transaction hash from path: %s", err)
+		c.JSON(400, fmt.Sprintf("invalid transaction hash: %s", err))
+		return
+	}
+
+	txHash := uriParams.TxHash
+	// convert to cbor bytes
+	cborData, err := cbor.Marshal(txHash)
+	if err != nil {
+		logger.Errorf("failed to encode transaction hash to CBOR: %s", err)
+		c.JSON(400, fmt.Sprintf("failed to encode transaction hash to CBOR: %s", err))
```

Comment:
Should this be a 500?

Reply:
I think no. Here's we're encoding the path given by the user, so a failure is always going to be because of bad input.